### DISCUSSION TITLE
How to remove decimals for Annotations display(?) and 2.6.0 quirk

### DISCUSSION THREAD

**Discussion by fatrabbit-la** [artisan-roaster-scope/artisan#854] [Q&A] [answered]
I use "G~E/20 @ ~Y2" to display time and gas settings along the way. Works great. Is there a way I can further customize this to get rid of the decimals and show only the whole number? This will help clean things up and make it easier to read.

Also, not to complicate things (this is a picky request) but is there a way in the formula to have the BT 'snap' to a desired temp? For example, most of my roasts target specific temps. Let's say one of them is 315F. By the time I click the button to mark the event, it's a bit past this. Not a big deal at all, but when I'm following a background roast, it's easier to see "315" for 315 rather than "317" or "318". I am not a programmer but I imagine something like, IF temp >=315 AND <320 then write 315.

Finally, I'm on 2.6.0 on my Mac laptop running OS 12.2.1. Artisan seems to be running just fine but I noticed over several roasts that the automatic CHARGE annotation came in around 15 degrees low. I know I can control-click at the actual charge to correct this but I haven't had to do this in the past.

**Comment by MAKOMO** [comment, marked as answer]
> I use "G~E/20 @ ~Y2" to display time and gas settings along the way. Works great. Is there a way I can further customize this to get rid of the decimals and show only the whole number? This will help clean things up and make it easier to read.

There is a flag to have all temperatures displayed without decimals, which  are mostly unproductive in roast profiles. Find it under `Config >> Curves`, last tab UI. The flag is named "Decimal Places".

> Also, not to complicate things (this is a picky request) but is there a way in the formula to have the BT 'snap' to a desired temp? For example, most of my roasts target specific temps. Let's say one of them is 315F. By the time I click the button to mark the event, it's a bit past this. Not a big deal at all, but when I'm following a background roast, it's easier to see "315" for 315 rather than "317" or "318". I am not a programmer but I imagine something like, IF temp >=315 AND <320 then write 315.

What about using an alarm to mark those events automatically or use the autoDRY feature which will mark it at the defined DRY end phase time.

> Finally, I'm on 2.6.0 on my Mac laptop running OS 12.2.1. Artisan seems to be running just fine but I noticed over several roasts that the automatic CHARGE annotation came in around 15 degrees low. I know I can control-click at the actual charge to correct this but I haven't had to do this in the past.

Nothing changed there. The autoCHARGE feature is a heuristic and on how precise it marks the event depends on the curve shape. Note that you can move the event easily at any time using a right click on the correct position. You can also zoom in to set the mark more precisely. Note further those marks can only placed on actual readings thus if you have a higher sampling rate your marks can be placed more precisely.

**Reply by fatrabbit-la** [reply]
Thank you for the solution to removing the decimals.

If I understand your suggestion to use an alarm to mark the events, I don't think that will work for me. Most of my profiles do follow the same gas @ BT changes but not always and I also do full batches, 3/4 batches, and half batches so they have different profiles. But that's ok - not that big a deal.

The autoCHARGE feature is curious. Yes, I'm aware of right-clicking to mark the correct position and I do that after the fact but I've been using Artisan for years and have literally never experienced this issue until this release. I say never when really I mean 99% of the time. This is a strange one. Nothing about my setup has changed unless the software defaulted to something different when I installed this latest version.
